x-pack/plugins/endpoint/public/applications/endpoint/view/hosts/details/components/flyout_sub_header.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo } from 'react';
+import { EuiFlyoutHeader, CommonProps, EuiButtonEmpty } from '@elastic/eui';
+import styled from 'styled-components';
+
+export type FlyoutSubHeaderProps = CommonProps & {
+  children: React.ReactNode;
+  backButton?: {
+    title: string;
+    onClick: (event: React.MouseEvent) => void;
+    href?: string;
+  };
+};
+
+const StyledEuiFlyoutHeader = styled(EuiFlyoutHeader)`
+  padding: ${props => props.theme.eui.paddingSizes.s};
+
+  &.hasButtons {
+    .buttons {
+      padding-bottom: ${props => props.theme.eui.paddingSizes.s};
+    }
+
+    .back-button-content {
+      padding-left: 0;
+      &-text {
+        margin-left: 0;
+      }
+    }
+  }
+
+  .flyout-content {
+    padding-left: ${props => props.theme.eui.paddingSizes.m};
+  }
+`;
+
+const BUTTON_CONTENT_PROPS = Object.freeze({ className: 'back-button-content' });
+const BUTTON_TEXT_PROPS = Object.freeze({ className: 'back-button-content-text' });
+
+/**
+ * A Eui Flyout Header component that has its styles adjusted to display a panel sub-header.
+ * Component also provides a way to display a "back" button above the header title.
+ */
+export const FlyoutSubHeader = memo<FlyoutSubHeaderProps>(
+  ({ children, backButton, ...otherProps }) => {
+    return (
+      <StyledEuiFlyoutHeader hasBorder {...otherProps} className={backButton && `hasButtons`}>
+        {backButton && (
```

Comment:
what is the purpose of having a second back button?

Reply:
There is only 1 back button - if you look at the screen capture above, you can see the link in the sub-panel. I will rename the variable to `backToDetails` in order to be clearer :)

Reply:
Also - FYI: I made this a separate component because I think we'll have more use cases we will need it - at least that seems to be true for Host details. Once we have that need, we should likely move the component to a higher directory.

Reply:
Sorry, I was thinking 'second' as in addition to the browser back button. does this do the same thing? or is it different

Reply:
@oatkiller 
> does this do the same thing? or is it different

Yes and No :)
The link (in the current usage by Host Policy Response panel) will always take you to the Host Details panel. If user navigated to the Policy Response panel from the host details, then yes, the behaviour would be the same as the browser back button. But since this panel is driven by a URL param, the user might have also gotten there by using the URL directly - in that case, this link would (possibly) not match the behavior of the browser's back button.